addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCCommonHandler.java

```diff
@@ -0,0 +1,279 @@
+    protected int getAddress(final String name) {
+        int address = INVALID;
+
+        logger.debug("Get address of {} LOGO! for block {} .", getLogoFamily(), name);
+
+        final int base = getBase(name);
+        if (isValid(name) && (base != INVALID)) {
+            final String block = name.split("\\.")[0];
+            if (Character.isDigit(block.charAt(1))) {
+                address = Integer.parseInt(block.substring(1));
+            } else if (Character.isDigit(block.charAt(2))) {
+                address = Integer.parseInt(block.substring(2));
+            } else if (Character.isDigit(block.charAt(3))) {
+                address = Integer.parseInt(block.substring(3));
+            }
+        } else {
+            logger.warn("Wrong configurated LOGO! block {} found.", name);
+        }
+
+        return address;
+    }
+
+    /**
+     * Calculate address offset for given block name.
+     *
+     * @param name Name of the data block
+     * @return Calculated address offset
+     */
+    protected int getBase(final String name) {
+        int base = INVALID;
+
+        final String family = getLogoFamily();
+        logger.debug("Get base address of {} LOGO! for block {} .", family, name);
+
+        final String block = name.split("\\.")[0];
+        if (isValid(name) && !block.isEmpty()) {
+            final Map<?, Layout> memory = LOGO_MEMORY_BLOCK.get(family);
+            if (Character.isDigit(block.charAt(1))) {
+                base = memory.get(block.substring(0, 1)).address;
+            } else if (Character.isDigit(block.charAt(2))) {
+                base = memory.get(block.substring(0, 2)).address;
+            } else if (Character.isDigit(block.charAt(3))) {
+                base = memory.get(block.substring(0, 3)).address;
+            }
+        }
+
+        return base;
+    }
+
+    /**
+     * Checks if thing handler is valid and online.
+     *
+     * @return True, if handler is valid and false otherwise
+     */
+    protected boolean isThingOnline() {
+        final Bridge bridge = getBridge();
+        Objects.requireNonNull(bridge, "PLCCommonHandler: Bridge may not be null.");
+
+        final Thing thing = getThing();
+        Objects.requireNonNull(thing, "PLCCommonHandler: Thing may not be null.");
+        return ((ThingStatus.ONLINE == bridge.getStatus()) && (ThingStatus.ONLINE == thing.getStatus()));
+    }
+
+    protected @Nullable State getOldValue(final String name) {
+        synchronized (oldValues) {
+            return oldValues.get(name);
+        }
+    }
+
+    protected void setOldValue(final String name, final @Nullable State value) {
+        synchronized (oldValues) {
+            oldValues.put(name, value);
+        }
+    }
+
+    /**
+     * Returns configured LOGO! communication client.
+     *
+     * @return Configured LOGO! client
+     */
+    protected PLCLogoClient getLogoClient() {
+        Objects.requireNonNull(client, "PLCCommonHandler: Client may not be null.");
```

Comment:
Why would you want this, please take a careful look at all of them.

Reply:
This communication client will be created by bridge during initialization. Nothing will work, if no communication client is available. To avoid multiple checks if client was constructed properly, simply check it in getter.

Reply:
Yes, but my point is that this will throw an NPE now, this is imho not the way to handle this, if this can happen you most likely cripple the binding at runtime, if it can't happen, then this is the wrong construct imho.

@wborn wdyt?

Reply:
I think the problem is when using `Objects.requireNonNull` everywhere is that the code will also start throwing NPEs everywhere. 

The calling code might not expect these NPEs to be thrown so in the end null values will not be properly handled. Annotating the method with `@Nullable` instead would make it clear where these null values need to be handled. 

The `PLCBridgeHandler.getLogoClient()` method does it the right way.